_source/_posts/2020-12-08-beginners-guide-to-jwt.adoc

```diff
@@ -0,0 +1,175 @@
+---
+layout: blog_post
+title: "A Beginner's Guide to JWTs"
+author: brian-demers
+by: advocate
+communities: [security]
+description: "What is a JWT? How are they used? Find out in this post!"
+tags: [security, tokens, jwt]
+tweets:
+- "Learn about the format of #JWT and how they are used to create secure applications🔒"
+- "JWTs are a key part of securing the web, learn more about them in this post📡"
+- "Still confused about what a #JWT is? this post will help‼️"
+image: blog/featured/okta-java-headphones.jpg
+type: awareness
+---
+:toc: macro
+:page-liquid:
+
+JSON Web Tokens (JWT) are used everywhere (even places they shouldn't be). This post will cover the basics of what you need to know about JWT and the related specifications in the **J**avascript **O**bject **S**igning and **E**ncryption (JOSE) family.
+
+TIP: JWT is pronounced "jot".
+
+toc::[]
+
+== What is a JWT
+
+A JWT is a structured security token format used to encode JSON data. Tokens can be cryptographically verified to ensure they have not been changed. There are two types of JWTs:
+
+- **J**SON **W**eb **S**ignature (JWS)
+- **J**SON **W**eb **E**ncryption (JWE)
+
+The data in a JWS is public, meaning anyone with the token can read the data, whereas a JWE is encrypted and private.  To read data contained within a JWE, you need both the token and a secret key.
```

Comment:
I think it would be instructive to give examples here. "A JWS can be trivially inspected using base64 decoding" ... etc